Add support for Apicurio Registry 2.x Avro library

Ladicek opened the pull request.

Ladicek committed:
Add support for Apicurio Registry 2.x Avro library

Ladicek commented:
Draft because Apicurio Registry 2.x hasn't been released yet.

gsmet commented on extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java:
```diff
@@ -103,6 +106,12 @@
 
     static final DotName OBJECT_MAPPER = DotName.createSimple("com.fasterxml.jackson.databind.ObjectMapper");
 
+    // TODO do we want this?
```
Yes we do and we can remove this comment :).

Ladicek replied:
Thanks! :-)

Ladicek commented:
And also I need to add a test :-)

gsmet reviewed (dismissed):
Looks good. I added a small comment, please dismiss when fixed

Ladicek committed:
Add tests for Apicurio Registry 2.x Avro library

Ladicek committed:
Update Kafka + Avro documentation to use Apicurio Registry 2.x

Ladicek commented:
To fix the test failure, I rebased my branch and added https://github.com/quarkusio/quarkus/pull/16251/files#diff-6d16e1bd2cec46a1c2d8f00ec4f3476b7554df6379ac769a0492f8aad9f69911 to the new `kafka-avro-apicurio2` test suite module.

Ladicek commented:
That looks quite unrelated.

Ladicek commented:
Sorry, I don't know how to properly "dismiss" the request for change :-)

Ladicek commented:
Ha, now I know how to dismiss the request :-) Haha, sorry for the noise!

I also rebased and re-pushed, maybe the unrelated flaky test will pass this time.

Ladicek commented:
Rebased, as #16746 might have fixed the MySQL issue.

cescoffier reviewed (approved):
Awesome!
